sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ProtoMessageDescriptors.scala

```diff
@@ -16,34 +16,110 @@
 
 package kalix.springsdk.impl
 
-import com.google.protobuf.DescriptorProtos
+import java.lang.reflect.AnnotatedParameterizedType
+import java.lang.reflect.Field
+import java.lang.reflect.ParameterizedType
+import java.time.Instant
+
+import scala.annotation.tailrec
+import scala.jdk.CollectionConverters._
+
 import com.fasterxml.jackson.dataformat.protobuf.ProtobufMapper
+import com.fasterxml.jackson.dataformat.protobuf.schema.ProtobufField
 import com.fasterxml.jackson.dataformat.protobuf.schema.ProtobufMessage
 import com.fasterxml.jackson.dataformat.protobuf.{ schema => jacksonSchema }
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.google.protobuf.DescriptorProtos
 import com.google.protobuf.DescriptorProtos.FieldDescriptorProto
 
-import scala.jdk.CollectionConverters._
-
 /**
  * Extracts a protobuf schema for a message, used only for assigning a typed schema to view state and results
  */
 object ProtoMessageDescriptors {
   private val protobufMapper = ProtobufMapper.builder.addModule(new JavaTimeModule).build;
 
   def generateMessageDescriptors(javaClass: Class[_]): ProtoMessageDescriptors = {
+
     val jacksonProtoSchema = protobufMapper.generateSchemaFor(javaClass)
 
     val messages = jacksonProtoSchema.getMessageTypes.asScala.toSeq.map { messageType =>
       val jacksonType = jacksonProtoSchema.withRootType(messageType).getRootType
-      toProto(jacksonType)
+      toProto(jacksonType, javaClass)
     }
     val (Seq(mainDescriptor), otherDescriptors) =
       messages.partition(_.getName.endsWith(jacksonProtoSchema.getRootType.getName))
+
     ProtoMessageDescriptors(mainDescriptor, otherDescriptors)
   }
 
-  private def toProto(jacksonType: ProtobufMessage): DescriptorProtos.DescriptorProto = {
+  private def isTimestampField(
+      jacksonType: ProtobufMessage,
+      protoField: ProtobufField,
+      javaClass: Class[_]): Boolean = {
+
+    def lookupOriginalJavaType(jacksonType: ProtobufMessage, javaClass: Class[_]): Option[Class[_]] = {
+
+      def isClassMatch(cls: Class[_]): Boolean = cls.getSimpleName == jacksonType.getName
+      def isFieldMatch(field: Field): Boolean = isClassMatch(field.getType)
+
+      if (isClassMatch(javaClass)) Some(javaClass)
+      else {
+        javaClass.getFields
```

Comment:
will this `.getFields` work if javaClass is a record? I think it will always return empty.. I think you'll need to use `getDeclaredFields` or `getRecordComponents`

Reply:
Yeah, I will check this. 

Btw, I converted one of the types to java class, namely the `TimeEnum` because IntelliJ was complaining. I will revert it back to record and may also convert some of the types we have in the tests to record.

Reply:
> I converted one of the types to java class, namely the `TimeEnum` because IntelliJ was complaining.

yeah, my intellij also compains about that one.. I wonder if we have any missing config

Reply:
I got it working by:

* configuring sbt to always use 17 for all the Spring SDK artifacts
* manually configure IntelliJ to use 17 for the Spring SDK modules